repaygithub/ui-tools
# CACTUS-253 :: Upgrade html-webpack-plugin

Ticket: https://repayonline.atlassian.net/browse/CACTUS-253

I didn't notice any changes we needed to make as far as the initial upgrade to v4.3.0 goes...all seemed to be working fine immediately after upgrading.

Let me know what you guys think about the CLI arg name/alias. I'm open to suggestions if anyone has one they'd like to put forward; I just put down the first thing that came to mind

Sorry the testing instructions are lengthy, but there are several things we should be double-checking

### Testing

1. Check out this branch
2. Run `yarn install` in this repo
3. Navigate to `modules/repay-scripts` in your terminal
4. Run `yarn link`
5. Now, go to the cactus repo and run `yarn link @repay/scripts` at the root.
6. Navigate to `examples/mock-ebpp` in your terminal
7. Run `yarn dev` and make sure you can still view the app as normal in the browser
8. Run `yarn build` and verify that you don't see any build errors in the console
9. Now, add a file in the `src` folder of `mock-ebpp` called `index.html` and paste the following contents in it:
```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <meta name="viewport" content="If this shows up, it worked" />
    <title><%= htmlWebpackPlugin.options.title %></title>
  </head>
  <body>
  </body>
</html>
```
10. Run `repay-scripts dev src/index.tsx -t src/index.html -p 3437` and open the app in your browser.
11. Inspect the HTML in the page and verify that you see the viewport meta tag with the content in the `index.html` file you set up.
12. Run `repay-scripts build src/index.tsx -t src/index.html`
13. Open `dist/index.html` and check that the viewport meta tag with the content in the `index.html` file you set up is there.
14. Run `yarn dev` & make sure you see the viewport meta contents from your `index.html` when you inspect the page
15. Run `yarn build` & make sure you see the viewport meta contents from your `index.html` in the `dist` folder
